### PR TITLE
lib/internal: fix vet errors, and error handling

### LINKED ISSUE #194: Go 1.7 vet compliance violation
Currently docker2aci copies the `http#DefaultTransport` struct, and modifies it , see https://github.com/appc/docker2aci/blob/deadf1f9d3a9ca2144efbb65ad223e535f8f9c24/lib/internal/util/util.go#L76.

This causes `go vet` starting from Go 1.7 to complain. In rkt we moved to using an internal client rather than the stdlib defaults.

xref https://github.com/coreos/rkt/pull/3084

### PR DESCRIPTION
**Pull request by s-urbaniak**
This introduces go vet, and fixes error handling in backend/file.

Fixes #194

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -30,4 +30,6 @@ if [ "${GOOS}" = "freebsd" ]; then
 fi
 
 echo "Building docker2aci..."
+go vet ./pkg/...
```
Comment (jonboulle) [inline build.sh]: can you put this in tests/tests.sh instead?

**Comment by s-urbaniak** [comment]
Unfortunately go 1.6.3 (the version we are building with on Semaphor) has a bug with struct literals, and slices, as reported in https://github.com/golang/go/issues/9171. So I am tempted to disable these `go vet` tests now at all :-/

Another possibility: we bump go 1.7 in Semphore.

File: build.sh (hunk 1 above)
Comment (s-urbaniak) [inline build.sh]: good question, personally I see those `go vet` commands actually part of the "compile" phase independent of any tests, but I can move it over. We have a problem with Go 1.6.3 though in Semaphore (see https://github.com/appc/docker2aci/pull/210#issuecomment-255387525).

**Comment by s-urbaniak** [comment]
Nevermind, I found a way to trick it.

**Comment by s-urbaniak** [comment]
@jonboulle addressed review comments, PTAL

**Comment by jonboulle** [comment]
LGTM, thanks!

**Pull request merged**